corehq/apps/users/views/__init__.py

```diff
@@ -1140,14 +1140,28 @@ def invite_web_user_form(self):
         role_choices = get_editable_role_choices(self.domain, self.request.couch_user, allow_admin_role=True)
         domain_request = DomainRequest.objects.get(id=self.request_id) if self.request_id else None
         is_add_user = self.request_id is not None
-        initial = {
-            'email': domain_request.email if domain_request else None,
-        }
+        invitation = self.invitation
+        if invitation:
+            assigned_location = invitation.assigned_locations.filter()
+            assigned_location_ids = [loc.location_id for loc in assigned_location]
+            primary_location_id = invitation.primary_location.location_id
+            initial = {
+                'email': invitation.email,
+                'role': invitation.role,
+                'assigned_locations': assigned_location_ids,
+                'primary_location': primary_location_id,
+            }
+        else:
+            initial = {
+                'email': domain_request.email if domain_request else None,
+            }
         can_edit_tableau_config = (self.request.couch_user.has_permission(self.domain, 'edit_user_tableau_config')
                                 and toggles.TABLEAU_USER_SYNCING.enabled(self.domain))
         if self.request.method == 'POST':
             current_users = [user.username for user in WebUser.by_domain(self.domain)]
             pending_invites = [di.email for di in Invitation.by_domain(self.domain)]
+            if invitation and invitation.email in pending_invites:
+                pending_invites.remove(invitation.email)
```

Comment:
I don't follow, what's going on here?

Reply:
The invite form will normally throw an error if you try to send an invite to an email that has an invite pending/already belongs to the domain. Since the edit form reuses that same form, I need to take out the email belonging to the invite that's being edited so it can submit/save properly.